lib/crypto/crypto_rsa.c

```diff
@@ -81,7 +82,7 @@ crypto_rsa_sign(int key, const uint8_t * data, size_t len,
 		goto err0;
 
 	/* Make sure the key and signature buffer are the correct size. */
-	if ((RSA_size(rsa) != 256) || (BN_num_bits(rsa->n) != 2048)) {
+	if (crypto_compat_RSA_valid_size(rsa)) {
```

Comment:
This would make more sense if the function returned nonzero for valid sizes and zero for invalid sizes, so that the code could be `if (!crypto_compat_RSA_valid_size(rsa)) { ...`.  Right now it reads as "if rsa has a valid size, warn that the size is incorrect".